wagtail/core/blocks/list_block.py

```diff
@@ -17,16 +17,26 @@
 
 
 class ListBlockValidationError(ValidationError):
-    def __init__(self, block_errors):
-        self.block_errors = block_errors
-        super().__init__('Validation error in ListBlock', params=block_errors)
+    def __init__(self, block_errors=None, non_block_errors=None):
+        self.non_block_errors = non_block_errors or ErrorList()
```

Comment:
On ``StreamBlockValidationError``, we don't set defaults here. So wonder why we need these for list block?

Reply:
I think StreamBlockValidationError is slightly in the wrong in this case - if non_block_errors isn't explicitly passed when constructing one, then the telepath adapter will fail when it tries to call `error.non_block_errors.as_data()`. That's not a big deal when validation errors are being thrown 99% of the time within StreamBlock.clean's own logic, but less ideal for things that are subclassing `clean`. (The fact that ListBlock's convention for block errors is slightly weirder - a list of mostly `None` values` - also means that letting user code avoid that detail is a good idea...)